Quarkus 0.14 problem with sourceDirectory specified in Pom

Hi,

I tried to run your example, but it seems like compilation is failing. 

```
[ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.3.21:compile (compile) on project kotlin-rest-http4k: Compilation failure: Compilation failure: 
[ERROR] /home/gandrian/Desktop/quarkus-quickstarts-master/kotlin-rest-http4k/src/main/kotlin/com/acme/model/JsonHelper.kt:[5,14] Unresolved reference: json

```

Is there something I am missing (all I did was `mvn compile`)? If not, could you update it so I can check about the second problem you mention?

Sorry I forget to commit the updated pom, now it should be ok.

I checked and same behaviour is with graalVM rc14 and rc16

Uberto

On Thu, 9 May 2019 at 07:18, Georgios Andrianakis <notifications@github.com>
wrote:

> Hi,
>
> I tried to run your example, but it seems like compilation is failing.
>
> [ERROR] Failed to execute goal org.jetbrains.kotlin:kotlin-maven-plugin:1.3.21:compile (compile) on project kotlin-rest-http4k: Compilation failure: Compilation failure:
> [ERROR] /home/gandrian/Desktop/quarkus-quickstarts-master/kotlin-rest-http4k/src/main/kotlin/com/acme/model/JsonHelper.kt:[5,14] Unresolved reference: json
>
>
> Is there something I am missing (all I did was mvn compile)? If not,
> could you update it so I can check about the second problem you mention?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/quarkusio/quarkus/issues/2367#issuecomment-490759017>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADVQDEPYWNVXVJKO56KOM3PUO6Z7ANCNFSM4HLPO5FA>
> .
>

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java
@@ -44,6 +44,7 @@ public class LocalProject {
 
     public static final String PROJECT_GROUPID = "${project.groupId}";
 
+    private static final String PROJECT_BASEDIR = "${project.basedir}";
     private static final String POM_XML = "pom.xml";
 
     public static LocalProject load(Path path) throws BootstrapException {
@@ -187,7 +188,8 @@ public class LocalProject {
 
     public Path getSourcesSourcesDir() {
         if (getRawModel().getBuild() != null && getRawModel().getBuild().getSourceDirectory() != null) {
-            return Paths.get(getRawModel().getBuild().getSourceDirectory());
+            String originalValue = getRawModel().getBuild().getSourceDirectory();
+            return Paths.get(originalValue.startsWith(PROJECT_BASEDIR) ? originalValue.replace(PROJECT_BASEDIR, this.dir.toString()) : originalValue);
         }
         return dir.resolve("src/main/java");
     }

